src/praisonai-agents/praisonaiagents/tools/decorator.py

```
@@ -229,9 +249,21 @@ def decorator(fn: Callable) -> FunctionTool:
+            retry_policy=retry_policy,
+            requires_approval=requires_approval
         )
-        
+
+        # Register approval requirement with the global registry so local,
+        # gateway, and served runs all honour this tool's human sign-off.
+        if tool_instance.risk_level is not None:
+            try:
+                from praisonaiagents.approval import add_approval_requirement
+                add_approval_requirement(tool_instance.name, tool_instance.risk_level)
+            except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)